server/src/test/java/org/apache/druid/server/coordinator/loading/HttpLoadQueuePeonTest.java

```diff
@@ -316,6 +333,30 @@ public void testLoadRateIsChangedWhenLoadSucceeds() throws InterruptedException
     );
   }
 
+  @Test
+  public void testBatchSize()
+  {
+    Assert.assertEquals(10, httpLoadQueuePeon.calculateBatchSize(SegmentLoadingMode.NORMAL));
+
+    httpLoadQueuePeon = new HttpLoadQueuePeon(
+        "http://dummy:4000",
+        MAPPER,
+        httpClient,
+        new HttpLoadQueuePeonConfig(null, null, null),
+        new WrappingScheduledExecutorService(
+            "HttpLoadQueuePeonTest-%s",
+            httpClient.processingExecutor,
+            true
+        ),
+        httpClient.callbackExecutor,
+        () -> SegmentLoadingMode.NORMAL,
+        new HistoricalLoadingCapabilities(1, 3)
+    );
```

Comment:
Why create another object? It seems to have the same arguments as the value already assigned to `httpLoadQueuePeon`.